# HG changeset patch
# User Sergey Mironov <grrwlf@gmail.com>
# Date 1379501803 -14400
#      Wed Sep 18 14:56:43 2013 +0400
# Node ID 923960c312bb9cdd74f8a31015c561d930e6104a
# Parent  c3119c263bd37497175d6b624ca4b9a7eebcf094
Enable the <script> tag

Ideally, I would be happy with the version of <script/> which forbids nesting. I
mean that I would like to allow only short form of the tag such as

  <script type={blessMime "foo/bar"} src={bless "http://url"} />

but not the long form

  <script type={blessMime "foo/bar"}>
	{user.Exploit}
  </script>

This way we could use blesses to double-check the safety and still allow
programmer to do some calculations with scripts. For example, I used it to
implement a basic dependency-checker among scripts. Unfortunately, I don't know
how to define a tag having only short form enabled.

Also, I am using the long form of <script> for another kind of hack.  Placing a
function call in it is the only way I know to hook the $(window).load() event of
jQuery, which is required by some jQuery widgets (NivoSlider is the example).

diff --git a/lib/ur/basis.urs b/lib/ur/basis.urs
--- a/lib/ur/basis.urs
+++ b/lib/ur/basis.urs
@@ -937,6 +937,8 @@
 
 val label : bodyTag ([For = id, Accesskey = string] ++ tableAttrs)
 
+val script : unit -> tag [Typ = mimeType, Src = url] head [] [] []
+
 
 (*** AJAX-oriented widgets *)
 
